rust/src/treefile.rs

```diff
@@ -465,20 +472,16 @@ impl Treefile {
     }
 
     /// Return the raw file descriptor for the postprocess script
-    pub(crate) fn get_postprocess_script_fd(&mut self) -> i32 {
-        self.externals
-            .postprocess_script
-            .as_mut()
-            .map_or(-1, raw_seeked_fd)
+    pub(crate) fn get_postprocess_script(&mut self) -> Option<&mut File> {
+        self.externals.postprocess_script.as_mut()
     }
 
-    pub(crate) fn get_add_file_fd(&mut self, filename: &str) -> i32 {
-        raw_seeked_fd(
-            self.externals
-                .add_files
-                .get_mut(filename)
-                .expect("add-file"),
-        )
+    /// Access the opened file object for the injected file
+    pub(crate) fn get_add_file(&mut self, filename: &str) -> &mut File {
+        self.externals
+            .add_files
+            .get_mut(filename)
+            .expect("add-file")
```

Comment:
I think this may be a leftover as we couldn't return a Result before, but now we can.

Reply:
OK I spent some time fighting borrowck on this.  Result IMO isn't right because this isn't a user error.  We're just using the string value as a lookup key that we own.

I tried to have a method return `Vec<(&str, &str, &mut File)>` so the outer loop could just use that, but lifetimes don't allow it; also we're trying to effectively iterate over two collections at once.  I am sure there are ways to make this work better like returning an iterator, but I don't want to block this whole series on revamping that.

I think keeping this as a panic is actually right, becuase it's a "shouldn't happen" case, not a "user typoed filename" case.

Reply:
Ack, thanks for the attempt and the report!